dnsapi/dns_omglol.sh

```diff
@@ -0,0 +1,363 @@
+ omgloladdress This is your omg.lol address, without the preceding @ - you can see your list on your dashboard at https://home.omg.lol/dashboard
+Issues: github.com/acmesh-official/acme.sh
+Author: @Kholin <kholin+omglolapi@omg.lol>  
+'
+
+#returns 0 means success, otherwise error.
+
+########  Public functions #####################
+
+# Please Read this guide first: https://github.com/acmesh-official/acme.sh/wiki/DNS-API-Dev-Guide
+
+#Usage: dns_myapi_add   _acme-challenge.www.domain.com   "XKrxpRBosdIKFzxW_CT3KLZNf6q0HG9i01zxXp5CPBs"
+dns_omglol_add() {
+  fulldomain=$1
+  txtvalue=$2
+  omglol_apikey="${omglol_apikey:-$(_readaccountconf_mutable omglol_apikey)}"
+  omglol_address="${omglol_address:-$(_readaccountconf_mutable omglol_address)}"
+
+  # As omg.lol includes a leading @ for their addresses, pre-strip this before save
+  omglol_address="$(echo "$omglol_address" | tr -d '@')"
+
+  _saveaccountconf_mutable omglol_apikey "$omglol_apikey"
+  _saveaccountconf_mutable omglol_address "$omglol_address"
+
+  _info "Using omg.lol."
+  _debug "Function" "dns_omglol_add()"
+  _debug "Full Domain Name" "$fulldomain"
+  _debug "txt Record Value" "$txtvalue"
+  _secure_debug "omg.lol API key" "$omglol_apikey"
+  _debug "omg.lol Address" "$omglol_address"
+
+  omglol_validate "$omglol_apikey" "$omglol_address" "$fulldomain"
+
+  dnsName=$(_getDnsRecordName "$fulldomain" "$omglol_address")
+  authHeader="$(_createAuthHeader "$omglol_apikey")"
+
+  _debug2 "  dns_omglol_add(): Address" "$dnsName"
+
+  omglol_add "$omglol_address" "$authHeader" "$dnsName" "$txtvalue"
+
+}
+
+#Usage: fulldomain txtvalue
+#Remove the txt record after validation.
+dns_omglol_rm() {
+  fulldomain=$1
+  txtvalue=$2
+  omglol_apikey="${omglol_apikey:-$(_readaccountconf_mutable omglol_apikey)}"
+  omglol_address="${omglol_address:-$(_readaccountconf_mutable omglol_address)}"
+
+  # As omg.lol includes a leading @ for their addresses, strip this in case provided
+  omglol_address="$(echo "$omglol_address" | tr -d '@')"
+
+  _info "Using omg.lol"
+  _debug fulldomain "$fulldomain"
+  _secure_debug ApiKey "$omglol_apikey"
+  _debug address "$omglol_address"
+
+  omglol_validate "$omglol_apikey" "$omglol_address" "$fulldomain"
+
+  dnsName=$(_getDnsRecordName "$fulldomain" "$omglol_address")
+  authHeader="$(_createAuthHeader "$omglol_apikey")"
+
+  omglol_delete "$omglol_address" "$authHeader" "$dnsName" "$txtvalue"
+}
+
+####################  Private functions below ##################################
+# Check that the minimum requirements are present.  Close ungracefully if not
+omglol_validate() {
+  omglol_apikey=$1
+  omglol_address=$2
+  fulldomain=$3
+
+  if [ "" = "$omglol_address" ]; then
+    _err "omg.lol base address not provided.  Exiting"
+    exit 1
+  fi
+
+  if [ "" = "$omglol_apikey" ]; then
+    _err "omg.lol API key not provided.  Exiting"
+    exit 1
```

Comment:
don't use `exit`.   use `return 1` and check the return value in the upper level

Reply:
Updated per comment

Reply:
@Neilpang ready for your re-review